anchore_engine/services/policy_engine/engine/vulns/cpe_matchers.py

```diff
@@ -0,0 +1,330 @@
+        ]
+        os_matches = self._scan_distro_packages_by_cpe(image)
+        all_matches = non_os_matches + os_matches
+
+        # Dedup between os and non-os? There could be name conflicts
+        deduped_matches = dedup_cpe_vulnerabilities(all_matches)
+        mapped = [
+            CpeMatch(image_cpe=img, vuln_cpe=vuln) for img, vuln in deduped_matches
+        ]
+        return mapped
+
+    def _scan_distro_packages_by_cpe(self, image: Image) -> List[CpeMatch]:
+
+        # Match the distro packages
+        with timer("os cpe matcher", log_level="debug"):
+            db = get_thread_scoped_session()
+            db.refresh(image)
+            os_matches = self._match_distro_packages_by_cpe(image)
+
+        return os_matches
+
+    def _query_cpe_matches(self, image_cpes: List[ImageCpe]) -> List[CpeMatch]:
+        """
+        Get the CVE ids for the input list of matches
+
+        :param db:
+        :param packages:
+        :return:
+        """
+        nvd_cpes = self.db_manager.query_nvd_cpe_matches(image_cpes, self.cpe_cls)
+        vulndb_cpes = self.db_manager.query_vulndb_cpes(image_cpes)
+        logger.debug(
+            "Found %s nvd matches and %s vulndb matches",
+            len(nvd_cpes),
+            len(vulndb_cpes),
+        )
+
+        # Join the found cpes against the input cpes, this done after the queries to make queries faster
+        nvd_findings = map_matches_to_image(nvd_cpes, image_cpes)
+        vulndb_findings = map_matches_to_image(vulndb_cpes, image_cpes)
+
+        return nvd_findings + vulndb_findings
+
+    @staticmethod
+    def get_cve_ids(matches: List[CpeMatch]) -> List[str]:
+        cve_ids = []
+
+        for match in matches:
+            if isinstance(match.vuln_cpe, VulnDBCpe):
+                cve_ids.extend(match.vuln_cpe.parent.referenced_cves)
+            else:
+                cve_ids.append(match.vuln_cpe.parent.name)
+
+        return list(set(cve_ids))  # Dedup the list
+
+    def _get_cpes_for_os_packages(
+        self, packages: List[ImagePackage]
+    ) -> List[Tuple[ImagePackage, ImageCpe]]:
+        return cpes_for_image_packages(packages)
+
+    def _match_distro_packages_by_cpe(self, image: Image) -> List[CpeMatch]:
+        """
+        Returns list of tuples of (imagecpe, vulncpe) that are matches
+
+        :param image:
+        :return: list of tuples
+        """
+        logger.spew(
+            "scanning os packages for cpe matches id=%s digest=%s",
+            image.id,
+            image.digest,
+        )
+
+        os_pkg_cpe_mappings = self._get_cpes_for_os_packages(image.packages)
+
+        logger.spew("distro cpes: %s", os_pkg_cpe_mappings)
+
+        os_cpes = [cpe for pkg, cpe in os_pkg_cpe_mappings]
+        # Get the matches
+        matches = self._query_cpe_matches(os_cpes)
+
+        logger.spew(
+            "pre-filter cpe distro findings: %s",
+            [(x.image_cpe, x.vuln_cpe.vulnerability_id) for x in matches],
```

Comment:
Similar to Sam's comment above, single-letter variables are harder to read.

Reply:
fixed